Modules/_io/textio.c

```diff
@@ -1142,14 +1291,31 @@ This also does an implicit stream flush.
 [clinic start generated code]*/
 
 static PyObject *
-_io_TextIOWrapper_reconfigure_impl(textio *self,
+_io_TextIOWrapper_reconfigure_impl(textio *self, PyObject *encoding,
+                                   PyObject *errors, PyObject *newline_obj,
                                    PyObject *line_buffering_obj,
                                    PyObject *write_through_obj)
-/*[clinic end generated code: output=7cdf79e7001e2856 input=baade27ecb9db7bc]*/
+/*[clinic end generated code: output=52b812ff4b3d4b0f input=671e82136e0f5822]*/
 {
     int line_buffering;
     int write_through;
-    PyObject *res;
+    const char *newline = NULL;
+
+    /* Check if something is in the read buffer */
+    if (encoding != Py_None || errors != Py_None) {
+        if (self->decoded_chars != NULL) {
+            _unsupported("It is not possible to set the encoding "
+                         "of a non seekable file after the first read");
```

Comment:
Why "non seekable"? Does it matter?

Reply:
Also, what happens if `newline` is changed while `self->decoded_chars` is not NULL? Should this all go in `textiowrapper_change_encoding`?

Reply:
"non seekable" is wrong.  I forgot to update it.
And yes, changing newline should be prohibited.